tests/test_parameter.py

```diff
@@ -37,9 +37,10 @@ def test_as_argument_returns_expected_parameter(self):
         assert arg.value_from.path == "abc"
 
     def test_as_input_returns_expected_parameter(self):
-        param = Parameter("a", default="42").as_input()
+        param = Parameter("a", default="42", enum=["42", "100"]).as_input()
```

Comment:
Might be helpful to add `if default not in enum: raise(Error)` in the `__init__` of Parameter?